src/StructuredLogger/StructuredLogger.csproj

```diff
@@ -28,6 +28,8 @@
     <PackageReference Include="Microsoft.Build.Framework" Version="$(MSBuildPackageVersion)" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MSBuildPackageVersion)" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="all" />
+    <PackageReference Include="MSBuild.BinlogRedactor.SensitiveDataDetector" Version="0.0.4-beta" />

```

Comment:
Same concern with adding another dependency here. Do you think it'd be possible to only have the `StructuredLogViewer` and `BinlogTool` projects depend on this one? So that people who only depend on StructuredLogger.dll do not inherit two more dependencies they probably don't need? This would mean that StructuredLogger.dll doesn't have the built-in ability to redact.

Reply:
I added a new indirection project (`StructuredLogger.Utils`) to separate the functionality from the `StructuredLogger`.
I wanted to avoid duplicating or linking the redaction work in both of the ui layers. (The existing `StructuredLogViewer.Core` is not suitable for binlogtool and referencing binlogtool from the viewer felt too unnecesary)

Reply:
Sounds good. Let's make sure all .dlls that need to ship with the viewer are mentioned here:
https://github.com/KirillOsenkov/MSBuildStructuredLog/blob/main/src/StructuredLogViewer/MSBuildStructuredLogViewer.nuspec

For binlogtool, I think the packing is all automatic, so nothing should be done hopefully.

Reply:
So for binlogtool I installed it from local built package and verified it works as expected - good.
For the viewer I verified by building, extracting the lib folder from the created packages and verifying it works as expected - not sure if there is better way of testing.